test/Microsoft.Extensions.FileSystemGlobbing.Tests/FunctionalTests.cs

```diff
@@ -338,6 +338,37 @@ public void SingleFileAndRecursive()
                 "src/project2/source1.cs");
         }
 
+        [Fact]
+        public void StemCorrectWithDifferentWildCards()
```

Comment:
Any match that had multiple sub directories under it would produce incorrect stems for the files in the sub directories